Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/Pointers/DefaultPrimaryPointerSelector.cs

```diff
@@ -0,0 +1,140 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System.Collections.Generic;
+
+namespace Microsoft.MixedReality.Toolkit.Input
+{
+    /// <summary>
+    /// Default primary pointer selector. The primary pointer is chosen among all interaction enabled ones using the following rules in order:
+    ///   1. Currently pressed pointer that has been pressed for the longest
+    ///   2. Pointer that was released most recently
+    ///   3. Pointer that became interaction enabled most recently
+    /// </summary>
+    public class DefaultPrimaryPointerSelector : IMixedRealityPrimaryPointerSelector
+    {
+        private readonly Dictionary<IMixedRealityPointer, PointerInfo> pointerInfos = new Dictionary<IMixedRealityPointer, PointerInfo>();
+
+        public void Initialize()
+        {
+            if (MixedRealityToolkit.InputSystem != null)
+            {
+                MixedRealityToolkit.InputSystem.PointerEvent += OnPointerEvent;
```

Comment:
Oh this is an interesting pattern! I've never seen it used in MRTK. I like it. When did we add ability to directly listen to pointer events?

Reply:
That would be PR #4536 , which is not in yet.